### PR TITLE
support for ppc64le

### PR DESCRIPTION
**Pull request by aklyachkin**
Signed-off-by: Andrey Klyachkin <aklyachkin@gmail.com>

## Description
Support for Linux/ppc64le

## Motivation and Context
- [X] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

## How Has This Been Tested?
succesfully compiled on RHEL7/ppc64le, running on Kubernetes cluster

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: watchdog/build.sh [hunk 1]
```diff
@@ -24,6 +24,7 @@ fi
 
 docker build --no-cache --build-arg PLATFORM="-armhf" -t openfaas/classic-watchdog:latest-dev-armhf . -f Dockerfile.packager
 docker build --no-cache --build-arg PLATFORM="-arm64" -t openfaas/classic-watchdog:latest-dev-arm64 . -f Dockerfile.packager
+docker build --no-cache --build-arg PLATFORM="-ppc64le" -t openfaas/classic-watchdog:latest-dev-ppc64le . -f Dockerfile.packager
```
Comment (alexellis) [inline watchdog/build.sh]: I think you're missing the part that pushes this change

File: auth/basic-auth/Dockerfile.ppc64le [hunk 2]
```diff
@@ -0,0 +1,34 @@
+FROM golang:1.10-alpine3.9 as build
```
Comment (alexellis) [inline auth/basic-auth/Dockerfile.ppc64le]: Is this identical? If so, do we need to maintain a separate Dockerfile?

**Comment by alexellis** [comment]
Hi @aklyachkin,

You are missing a step here -> https://github.com/openfaas/faas/blob/master/.travis.yml#L21

It publishes the Docker image for the multi-arch watchdog.

Please could you squash down the commits after adding this?

Alex

File: auth/basic-auth/Dockerfile.ppc64le (hunk 2 above)
Comment (aklyachkin) [reply auth/basic-auth/Dockerfile.ppc64le]: Yes, it is identical. It seems golang is a multiarch image. As for now the only real difference is:

```
16c16
<     go test $(go list ./... | grep -v /vendor/) -cover
---
>     CGO_ENABLED=0 GOOS=linux go test $(go list ./... | grep -v /vendor/) -cover
```

I'm setting the variables CGO_ENABLED and GOOS for go test.

**Comment by aklyachkin** [comment]
Hi Alex,

thanks for pointing it out. I changed .travis.yml and squashed the commits.

File: auth/basic-auth/Dockerfile.ppc64le (hunk 2 above)
Comment (alexellis) [reply auth/basic-auth/Dockerfile.ppc64le]: We could pass that as an ARG instead and then remove the additional file?

`CGO_ENABLED=0` for `go test` should work for the main x86_64 build. Can you try it?

**Comment by burtonr** [comment]
Hey @aklyachkin have you had a chance to address Alex's comment about the Dockerfile `ARG` in order to prevent duplicating the files?

It seems that's the last outstanding change request here. Let us know if you need any help with it

**Comment by aklyachkin** [comment]
Hi @burtonr. No unfortunately I still didn't have to time to check it.

File: auth/basic-auth/Dockerfile.ppc64le (hunk 2 above)
Comment (jayapriya90) [reply auth/basic-auth/Dockerfile.ppc64le]: Looks like there's no activity in this PR for a month. I can try this change if it's fine.

**Comment by alexellis** [comment]
Closing as it seems that @aklyachkin isn't working on this anymore, we're in Feb and changes were requested in early Sep 2019.

**Pull request closed without merging**